packages/core/src/components/Form.tsx

```diff
@@ -453,12 +453,14 @@ export default class Form<
    * @param formData - The data for the `Form`
    * @param fields - The fields to keep while filtering
    */
-  getUsedFormData = (formData: T, fields: string[]): T => {
+  getUsedFormData = (formData: T, fields: string[][]): T => {
     // For the case of a single input form
     if (fields.length === 0 && typeof formData !== "object") {
       return formData;
     }
 
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore _pick has incorrect type definition, it works with string[][], because lodash/hasIn supports it
     const data: GenericObjectType = _pick(formData, fields);
```

Comment:
If you add the _pick<GenericObjectType>(formData, fields) does eliminate the need for the two comments above? or Perhaps a type cast?

Reply:
Here is type casting
```typescript
const data: GenericObjectType = _pick(formData, fields as unknown as string[]);
```

Do you think it would be better this way?

Reply:
Yes, that works. Please make sure to comment about the cast, otherwise, looks good.